dev-cpp/sol2/sol2-3.3.0.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 2021-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-{1,3,4} luajit )
+inherit cmake lua
+
+DESCRIPTION="Header-only C++ <-> Lua API wrapper"
+HOMEPAGE="https://github.com/ThePhD/sol2"
+SRC_URI="https://github.com/ThePhD/sol2/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+REQUIRED_USE="test? ( ${LUA_REQUIRED_USE} )"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	test? (
+		${LUA_DEPS}
+		>=dev-cpp/catch-3
+	)
+"
+BDEPEND="
+	test? (
+		virtual/pkgconfig
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/sol2-3.3.0-werror.patch
+	"${FILESDIR}"/sol2-3.2.2-luajit-pkgconf.patch
+	"${FILESDIR}"/sol2-3.3.0-catch-depend.patch
+	"${FILESDIR}"/sol2-3.3.0-cmake-dir.patch
+	"${FILESDIR}"/sol2-3.3.0-dont-install-tests.patch
+)
+
+src_prepare() {
+	cmake_src_prepare
+}
+
+src_configure() {
+	sol2_configure_wrapper() {
+		local mycmakeargs=(
+			-DSOL2_BUILD_LUA=no
+			-DSOL2_TESTS=$(usex test)
+			-DSOL2_LUA_VERSION="${ELUA}"
+		)
+		cmake_src_configure
+	}
+	lua_foreach_impl sol2_configure_wrapper
+}
+
+src_compile() {
+	lua_foreach_impl cmake_src_compile
+}
+
+src_install() {
+	lua_foreach_impl cmake_src_install
+}
```

Comment:
Is the lua_foreach_impl useful? Doesn't seem to install per implementation different files.

Either way, if keep src_install, I'd move it below src_test to match execution order.

Reply:
In previous ebuild, there was a lua-free build directory which was used for install. + per-lua directory for tests

Now it requires lua to be found even if tests are disabled, so instead "any" of these dirs will work. But since there's no concept of any, let it be "all", as they install the same files anyway